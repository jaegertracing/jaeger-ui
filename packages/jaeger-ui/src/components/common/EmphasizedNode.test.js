// Copyright (c) 2020 The Jaeger Authors.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
// http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

import React from 'react';
import { render } from '@testing-library/react';
import EmphasizedNode from './EmphasizedNode';
import '@testing-library/jest-dom';

describe('<EmphasizedNode>', () => {
  const defaultProps = {
    height: 100,
    width: 500,
  };

  it('renders with default props', () => {
    const { container } = render(<EmphasizedNode {...defaultProps} />);

    const recElements = container.querySelectorAll('rect');

    expect(recElements.length).toBe(4);
    recElements.forEach(element => {
      expect(element).toHaveAttribute('height', defaultProps.height.toString());
      expect(element).toHaveAttribute('width', defaultProps.width.toString());
    });
  });
});
