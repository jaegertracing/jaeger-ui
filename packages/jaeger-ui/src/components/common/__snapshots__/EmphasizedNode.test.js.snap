// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<EmphasizedNode> renders with default props 1`] = `
<Fragment>
  <rect
    className="EmphasizedNode--contrast is-non-scaling"
    height={100}
    vectorEffect="non-scaling-stroke"
    width={500}
  />
  <rect
    className="EmphasizedNode--contrast is-scaling"
    height={100}
    width={500}
  />
  <rect
    className="EmphasizedNode is-non-scaling"
    height={100}
    vectorEffect="non-scaling-stroke"
    width={500}
  />
  <rect
    className="EmphasizedNode is-scaling"
    height={100}
    width={500}
  />
</Fragment>
`;
